src/lib/registration/migration_repositories.rb

```diff
@@ -17,67 +17,90 @@
 require "registration/sw_mgmt"
 
 module Registration
-  # this class activates the migration repositories
+  # this class activates the migration services and repositories
   class MigrationRepositories
     include Yast::Logger
 
     Yast.import "Pkg"
 
+    # reset the libzypp migration setup
     def self.reset
-      log.info "Resetting upgrade repos config"
-      repos = Yast::Pkg.GetUpgradeRepos()
-      repos.each { |repo| Yast::Pkg.RemoveUpgradeRepo(repo) }
+      # reset the solver
+      Yast::Pkg.SetSolverFlags("reset" => true)
 
       # deselect all pre-selected packages or patches
       Yast::Pkg.PkgReset
     end
 
-    attr_accessor :repositories, :install_updates
+    attr_accessor :repositories, :services, :install_updates
 
     def initialize
       @repositories = []
+      @services = []
       # install updates by default
       @install_updates = true
     end
 
-    # use the repositories from a libzypp service for the migration
-    # @param [String] service_name name of the service
-    def add_service(service_name)
-      service_repos = SwMgmt.service_repos(service_name)
-      repositories.concat(service_repos)
+    # does any configured service contain an update repo?
+    # @return [Boolean] true if at least one service repository is an update
+    #   repository
+    def service_update_repo?
```

Comment:
yeah, I get it from code, but method name is not much communicative...if I see below `MigrationRepositories.service_update_repo?` then it is hard to quess what it do